File: Questions/LC Questions/Math/2566. Maximum Difference by Remapping a Digit/solution.java
// TC: O(n), SC: O(n) where n is the number of digits in num
class Solution {
    public int minMaxDifference(int num) {
        String s = String.valueOf(num);

        // For MAX, replace the first non-9 digit with 9
        String maxVal = s;
        for (char digit : s.toCharArray()) {
            if (digit != '9') {
                maxVal = s.replace(digit, '9');
                break;
            }
        }

        // For MIN, replace the very first digit with 0
        char digitToChange = s.charAt(0);
        String minVal = s.replace(digitToChange, '0');

        return Integer.parseInt(maxVal) - Integer.parseInt(minVal);
    }
}

File: Questions/LC Questions/Math/2566. Maximum Difference by Remapping a Digit/solution.py
'''
    Explanation:
        To get the MAX value: Change the first non-9 digit to a 9
        To get the MIN value: Change the very first digit to a 0

        TC: O(n)
        SC: O(n) where n is the number of digits in num
'''

class Solution:
    def minMaxDifference(self, num: int) -> int:
        s = str(num)

        max_val = s
        for digit in s:
            if digit != '9':
                max_val = s.replace(digit, '9')
                break
        
        digit_to_change = s[0]
        min_val = s.replace(digit_to_change, '0')

        return int(max_val) - int(min_val)